docs/source/model/dependencies.rst

```diff
@@ -326,6 +326,44 @@ Then MLflow will save ``utils.py`` under ``code/`` directory in the model direct
 When MLflow loads the model for serving, the ``code`` directory will be added to the system path so that you can use the module in your model
 code like ``from utils import my_func``. You can also specify a directory path as ``code_paths`` to save multiple files under the directory:
 
+Use of ``code_paths`` Option for a Custom Library
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To include custom libraries or libraries from a private mirror server when logging your model, use the ``code_paths`` option. 
+This option allows you to upload .whl files or other dependencies alongside your model, ensuring all required libraries are available during serving.
+
+::
+
+    my_project/
+    |── train.py
+    └── custom_package.whl
+
+Then the following code can log your model with the custom package:
+
+.. code-block:: python
+    :caption: train.py
+
+    import mlflow
+    from custom_package import my_func
+
+
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input):
+            x = my_func(model_input)
+            # .. your prediction logic
+            return prediction
+
+
+    # Log the model
+    with mlflow.start_run() as run:
+        mlflow.pyfunc.log_model(
+            python_model=MyModel(),
+            artifact_path="model",
+            extra_pip_requirements=["code/custom_package.whl"],
```

Comment:
I think this is not the recommended way.

I recommend to upload the private package wheel file to an internal server or clould storage with a downloading link,
then when logging model, log extra_pip_requirements=[<wheel_link>]

Reply:
Within the note block showing an example object store path for 'once you're ready for production' is ideal, while the example you have here is perfect for dev testing while building a solution.

Reply:
Thank you for the comments both, I'll add the disclaimer.

Reply:
@WeichenXu123 @BenWilson2 I've added the note section, can you take another look?

Reply:
<img width="1196" alt="image" src="https://github.com/user-attachments/assets/c8f8ab7d-05e6-45cd-9704-591941b86ffa">